fbzmq/async/AsyncSignalHandler.h

```diff
@@ -92,6 +98,10 @@ class AsyncSignalHandler {
 
   // registered signals
   sigset_t registeredSignals_;
+
+#ifdef IS_BSD
+  std::unordered_map<int, sig_t> signalHandlers_;
```

Comment:
Use an array instead of map. Index of array would indicate signal-type and value would indicate signal-value. Map could get very expensive (i.e. one lookup for every message received).

signal.h has macros for MAX signal that you can leverage & initialize to 0.

Reply:
I had to tweak a little bit, since SIGRTMAX is linux specific. But BSD has the same concept.